src/datatrove/pipeline/filters/url_filter.py

```diff
@@ -69,11 +70,16 @@ def download_data(self):
         if self._downloaded:
             return
         download_dir = cached_assets_path(library_name="datatrove", namespace="filters", subfolder="url_filter")
-        if not os.path.isfile(os.path.join(download_dir, "adult", "domains")) or not os.path.isfile(
-            os.path.join(download_dir, "adult", "urls")
+
+        with (
+            InterProcessLock(os.path.join(download_dir, "url_filterblacklists.tar.gz.lock")),
+            tarfile.open(os.path.join(ASSETS_PATH, "url_filterblacklists.tar.gz"), "r:gz") as tar,
```

Comment:
do we need to open the tarfile before checking if the files already exist?

Reply:
valid point